src/hotspot/share/classfile/systemDictionary.hpp

```diff
@@ -105,12 +105,12 @@ class SystemDictionary : AllStatic {
   // Resolve a superclass or superinterface. Called from ClassFileParser,
   // parse_interfaces, resolve_instance_class_or_null, load_shared_class
   // "class_name" is the class whose super class or interface is being resolved.
-  static InstanceKlass* resolve_super_or_fail(Symbol* class_name,
-                                              Symbol* super_name,
-                                              Handle class_loader,
-                                              Handle protection_domain,
-                                              bool is_superclass,
-                                              TRAPS);
+  static InstanceKlass* resolve_with_circularity_detection(Symbol* class_name,
```

Comment:
Preceding comment needs updating if this is not necessarily for super class/interface.